gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt

```diff
@@ -146,18 +146,26 @@ internal class AvatarPickerViewModel(
             when (val result = avatarRepository.uploadAvatar(email, uri)) {
                 is GravatarResult.Success -> {
                     fileUtils.deleteFile(uri)
-                    _uiState.update { currentState ->
-                        val avatar = result.value
-                        currentState.copy(
-                            uploadingAvatar = null,
-                            emailAvatars = currentState.emailAvatars?.copy(
-                                avatars = buildList {
-                                    add(avatar)
-                                    addAll(currentState.emailAvatars.avatars.filter { it.imageId != avatar.imageId })
-                                },
-                            ),
-                            scrollToIndex = null,
-                        )
+                    val isAutoSelected = _uiState.value.emailAvatars?.selectedAvatarId == null
+                    if (isAutoSelected) {
+                        _uiState.update { AvatarPickerUiState(email, avatarPickerContentLayout) }
+                        refresh()
```

Comment:
I think it's completely reasonable to refresh the content like you said, to future proof in case of backend changes, but I wonder if we really have to clean the whole state?

Could we do this refresh action as part of the uploading indicator? With the current code, we hide the whole list, the profile card is recomposed fully and that seems like a lot to select an avatar.

Reply:
> Could we do this refresh action as part of the uploading indicator? With the current code, we hide the whole list, the profile card is recomposed fully and that seems like a lot to select an avatar.

I don't think it's easy to keep the uploading indicator until the `fetchAvatars` finishes, at least reusing the current methods, but I'll try to improve it.

Reply:
Correct, you have to use suspend functions that won't start a new coroutine. We do have one like that to [fetch avatars](https://github.com/Automattic/Gravatar-SDK-Android/blob/trunk/gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt#L209) but not for the profile.

Reply:
Done!